packages/dds/matrix/src/matrix.ts

```diff
@@ -536,20 +539,31 @@ export class SharedMatrix<T = any>
 			SnapshotPath.cols,
 			this.cols.summarize(this.runtime, this.handle, serializer),
 		);
+		const cellsSnapshot = this.cells.snapshot();
+		const props: Record<string, number> = {
+			csSize: cellsSnapshot.length, // cellsSnapshotSize
+			rc: this.rowCount, // rowCount
+			cc: this.colCount, // colCount
+		};
 		const artifactsToSummarize = [
-			this.cells.snapshot(),
+			cellsSnapshot,
 			this.pending.snapshot(),
 			this.setCellLwwToFwwPolicySwitchOpSeqNumber,
 		];
 
 		// Only need to store it in the snapshot if we have switched the policy already.
 		if (this.setCellLwwToFwwPolicySwitchOpSeqNumber > -1) {
-			artifactsToSummarize.push(this.cellLastWriteTracker.snapshot());
+			const trackerMatrixSnapshot = this.cellLastWriteTracker.snapshot();
+			artifactsToSummarize.push(trackerMatrixSnapshot);
+			props.tmsSize = trackerMatrixSnapshot.length; // trackerMatrixSnapshotSize
 		}
 		builder.addBlob(
 			SnapshotPath.cells,
 			serializer.stringify(artifactsToSummarize, this.handle),
 		);
+		if (telemetryContext?.push !== undefined) {
+			telemetryContext.push("fluid:SharedMatrix", "details", props);
+		}
```

Comment:
Are you checking `push` for `undefined` for some back-compat case? I guess old ContainerRuntime, new DataStoreRuntime?

I prefer we codify this in the interface itself, declaring `push` to be optional... but since it's a public interface, it may be stuck as optional until `3.0`? That's fine I guess.  Or if it goes into rc.2.0 as optional maybe in 2.0.0 it can be required. Don't spend a lot of time sorting it out, but do consider making it optional on the interface to get the compiler's help (no one else using it will think to check for `undefined` first otherwise!)

You can in-line the `undefined` check on the function like this:
```suggestion
		telemetryContext?.push?.("fluid:SharedMatrix", "details", props);
```